docs/autogen/templates/getting_started.rst.jinja

```diff
@@ -0,0 +1,726 @@
+required linker flags:
+
+.. literalinclude:: pkg_config_00.cmd
+   :language: bash
+
+This will output something like the output below, it will vary depending on the
+features enabled/disabled.
+
+.. literalinclude:: pkg_config_00.out
+   :language: bash
+
+You can pass the arguments above to your compiler, or using pkg-config like so:
+
+.. literalinclude:: hello_00.cmd
+   :language: bash
+
+.. note:: You do not need to link with SPDK/DPDK, as these are bundled
+  with **xNVMe**. However, do take note of the linker flags surrounding
+  ``-lxnvme``, these are required as SPDK makes use of
+  ``__attribute__((constructor))``. Without the linker flags, none of the SPDK
+  transports will work, as **ctors** will be "linked-out", and **xNVMe** will
+  give you errors such as **device not found**.
+
+Also, **xNVMe** provides three different libraries, two static, and a shared
+library as well. Here is what the different libraries are intended for:
+
+* ``libxnvme.a``, this is static library version of **xNVMe** and it comes with
+  **batteries included**, that is, all the third-party libraries are bundled
+  within the static library ``libxnvme.a``. Thus you only need to link with
+  **xNVMe**, as described above, and need not worry about linking with SPDK,
+  liburing etc.
+* ``libxnvme-static.a``, this is the same as ``libxnvme.a`` except it does
+  **not** come with **batteries included**, so you have to manually link with
+  SPDK, liburing, etc.
+* ``libxnvme-shared.so``, this is the shared library version of **xNVMe**, this
+  also does **not** come with **batteries included**, so when linking with, or
+  dynamically loading, the shared library version of **xNVMe** then you also
+  have to link or load the third-party dependencies.
+
+Using ``libxnvme.a`` is the preferred way to consume **xNVMe** as it comes with
+the correct version of the various third-party libraries and provides for
+a simpler link-target.
+
+Run!
+----
+
+.. literalinclude:: hello_01.cmd
+   :language: bash
+
+.. literalinclude:: hello_01.out
+   :language: bash
+
+This should conclude the getting started guide of **xNVMe**, go ahead and
+explore the :ref:`sec-tools`, :ref:`sec-c-api`, and :ref:`sec-examples`.
+
+Should **xNVMe** or your system still be misbehaving, then take a look in the
+:ref:`sec-troubleshooting` section or reach out by raising an issue_, start an
+asynchronous discussion_, or go to Discord_ for synchronous interaction.
+
+.. _sec-troubleshooting:
+
+Troubleshooting
+===============
+
+User space
+----------
+
+In case you are having issues using running with SPDK backend / using then make
+sure you following the config section
+:ref:`sec-gs-system-config-userspace-config` and if issues persist a solution
+might be found in the following subsections.
+
+No devices found
+~~~~~~~~~~~~~~~~
+
+When running ``xnvme enum`` then the output-listing is empty, there are no
+devices. When running with ``vfio-pci`` then this can occur when your devices
```

Comment:
When running with ``vfio-pci`` then this -> When running with ``vfio-pci``, this